### PR TITLE
feat(prometheus): Docker Daemon scraping for monitoring

### PR DESCRIPTION
**Pull request by 24kushang**
#### Summary:
This PR works on adding the functionality of monitoring Docker daemon using **Open Telemetry Collector service.** This would allow the monitoring of Docker Daemon and hence giving system statistics.

#### Changes:

1. **New Service in otel-collector config**
   - Introduced a new `service` named metrics which generates the docker daemon.
   - This connects to `/var/run/docker.sock` file and generates the docker service metrics.
2. **Port Exposed**
   - Exposed **`9464`** port, so that Prometheus can scrape the metrics and can be further used in creating dashborads.

**Contributor:**
@24kushang 

------

I need some guidance in connecting using the internal docker network.

### CODE REVIEW

File: prometheus/prometheus.yml [hunk 1]
```diff
@@ -2,4 +2,4 @@ scrape_configs:
   - job_name: "prometheus"
     scrape_interval: 5s
     static_configs:
-      - targets: ["prosody:5280","jvb:8080","jicofo:8888"]
+      - targets: ["prosody:5280","jvb:8080","jicofo:8888","otel:9464","172.21.0.2:9464"]
```
Comment (ltorje-8x8) [inline prometheus/prometheus.yml:5]: why do you have an IP address here?

**Review by aaronkvanmeerten — APPROVED** [review]

**Comment by 24kushang** [comment]
image
Screenshot of the `docker_statistics` dashboard

**Review by ltorje-8x8 — APPROVED** [review]

**Comment by ltorje-8x8** [comment]
Maybe it would be a good idea to create a readme under https://github.com/jitsi/docker-jitsi-meet/tree/master/prometheus similar to https://github.com/jitsi/docker-jitsi-meet/blob/master/log-analyser/README.md

**Comment by 24kushang** [comment]
Would add the `README.md` in the next commit in the same PR.

**Review by ltorje-8x8 — APPROVED** [review]

File: prometheus/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+# Prometheus Scraping & Grafana Dashboard for Jitsi
+
+## Overview
+
+This project aims to integrate **Prometheus** and **Grafana** with Jitsi to monitor and visualize performance metrics. It enhances Jitsi’s operational efficiency by providing detailed insights into resource utilization, network traffic, and service availability.
```
Comment (bgrozev) [inline prometheus/README.md:5]:
```suggestion
This project aims to integrate **Prometheus** and **Grafana** with Jitsi to monitor and visualize performance metrics. 
```

File: prometheus/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+- Docker
+- Docker Compose
+
+### Steps
+
+1. **Clone the Repository**
+
+   ```bash
+   git clone https://github.com/yourusername/prometheus-scraper-jitsi-meet.git
```
Comment (bgrozev) [inline prometheus/README.md:24]: What's this?

File: prometheus/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+2. **Setup Jitsi with Docker Compose**
+
+   Follow the [Jitsi Docker](https://github.com/jitsi/docker-jitsi-meet) setup instructions. <br>
+   Also, you could follow Self - Hosting guide of Jitsi Meet: [Jitsi handbook](https://jitsi.github.io/handbook/docs/devops-guide/devops-guide-docker/)
+
+3. **Deploy Prometheus and Grafana**
+
+   Adding the service in different files which allows for better user control. These files are added at root level.<br>
+
+   Add Prometheus service to your `prometheus.yml` which turns up the container:
```
Comment (bgrozev) [inline prometheus/README.md:37]: Is this really what the user is supposed to do to enable prometheus? I thought that's already part of prometheus.yml and the user simply needs to add `-f prometheus.yml` to the docker-compose commain

File: prometheus/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+   ```bash
+   docker-compose -f docker-compose.yml -f prometheus.yml -f grafana.yml -f log-analyser.yml up -d
+   ```
+
+## Usage
+
+1. **View the Prometheus Targets**
+
+   Open [http://localhost:9090](http://localhost:90990) in your browser.
```
Comment (bgrozev) [inline prometheus/README.md:94]:
```suggestion
   Open [http://localhost:9090](http://localhost:9090) in your browser.
```

**Review by bgrozev — APPROVED** [review]

**Pull request merged**